ofrak_core/test_ofrak/unit/test_resource.py

```diff
@@ -170,3 +172,61 @@ async def test_get_most_specific_tags(resource: Resource):
 
     expected_most_specific_tags = {GenericText, FilesystemRoot, Elf}
     assert set(resource.get_most_specific_tags()) == expected_most_specific_tags
+
+
+@pytest.mark.asyncio
+async def test_flush_to_disk_pack(ofrak_context: OFRAKContext):
+    # This test works as long as LZMA fails to pack() and the default is to
+    # pack recursively. The root resource is a LZMA archive in a LZMA archive,
+    # which as of 2022-09-13 will fail to pack and therefore pack_recursively
+    # won't work.
+
+    root_resource = await ofrak_context.create_root_resource(
+        "lzma",
+        b"\x5d\x00\x00\x80\x00\xff\xff\xff\xff\xff\xff\xff\xff\x00\x2e\x80\x2d\x00\x00\x7f\xeb\x0b\xed\x6e\xd4\x23\x99\xb4\xf3\x0e\x8d\x71\xaf\xff\xff\xf9\xbe\x40\x00",
```

Comment:
Black doesn't know how to break up long strings – this needs to be done manually. There are probably a bunch of long strings elsewhere in the codebase.

In this case, is there a reason to inline the file like this as opposed to LFS tracking a file that can be reused for other tests?

Reply:
I was inspired by another test that did a similar thing and liked the fact that it's obvious what bytes are operated on, especially when something fails and it can be compared to the output. Not sure how much use we have for an empty LZMA in LZMA, but I don't mind checking in a separate file. If we keep it, I can break it up to fit into 120 chars.

Reply:
I think let's go with a small, lfs-tracked file in an assets/ directory for this one. Thanks @sjossi.

Reply:
I'm creating a mock resource, since we don't rely on (or even want to use) LZMA anymore.